zalo_sdk/oa/Client.py

```diff
@@ -27,20 +27,24 @@ def check_zalo_error(self, response):
                 raise zalo_sdk.oa.ZaloOAException(response["error"], response["message"])
             raise zalo_sdk.oa.ZaloOAException.ZaloOAException(response["error"])
 
-    def request_authoriation_code_url(self, callback_url, challenge_string=None):
+    def request_authoriation_code_url(self, callback_url, code_challenge=None, state=None):
         """
         Get the URL to request the authorization code.
 
-        Official Documentation: https://developers.zalo.me/docs/api/official-account-api/xac-thuc-va-uy-quyen/cach-1-xac-thuc-voi-giao-thuc-oauth/lay-oa-access-token-tu-oa-refresh-token-post-4970
+        Official Documentation:
+        https://developers.zalo.me/docs/api/official-account-api/xac-thuc-va-uy-quyen/cach-1-xac-thuc-voi-giao-thuc-oauth/yeu-cau-cap-moi-oa-access-token-post-4307
+        https://developers.zalo.me/docs/api/official-account-api/xac-thuc-va-uy-quyen/cach-1-xac-thuc-voi-giao-thuc-oauth/lay-oa-access-token-tu-oa-refresh-token-post-4970
         """
         base_url = "https://oauth.zaloapp.com/v4/oa/permission"
         parsed_url = urllib.parse.urlparse(base_url)
         params = {
             'app_id': self.app_id,
             'redirect_uri': callback_url,
         }
-        if challenge_string is not None:
-            params['state'] = challenge_string
+        if code_challenge is not None:
+            params['code_challenge'] = code_challenge
+        if state is not None:
+            params['state'] = state
```

Comment:
I think we can set state = zalo_connection_id for easier to find back
then decode the code_challenge to check with code_verifier

Reply:
According to this doc, we need to generate a unique and random `state`: https://www.oauth.com/oauth2-servers/pkce/authorization-request/

Reply:
hm... the id in odoo is not random, maybe we can generate another uuid for that